spec/features/staff/users_can_sign_in_spec.rb

```diff
@@ -42,34 +42,140 @@ def log_in_via_form(user, remember_me: false)
+
+        # When I enter the one-time password before it expires
+        travel 3.minutes do
+          fill_in "Please enter your six-digit verification code", with: otp_at_time_of_login
+          click_on "Continue"
+        end
+
+        # Then I should be logged in
+        expect(page).to have_link(t("header.link.sign_out"))
+        expect(page).to have_content("Signed in successfully.")
+
+        # And I should be at the home page
+        expect(page).to have_content("You can search by RODA, Delivery Partner, or BEIS identifier, or by the activity's title")
+      end
 
-      # When I log in with that user's email and password
-      visit root_path
-      log_in_via_form(user)
+      scenario "unsuccessful OTP attempt" do
+        # Given a user with 2FA enabled and a confirmed mobile number exists
+        user = create(:administrator, :mfa_enabled, mobile_number_confirmed_at: DateTime.now)
+
+        # When I log in with that user's email and password
+        visit root_path
+        log_in_via_form(user)
 
-      otp_at_time_of_login = user.current_otp
-      # Then I should receive an OTP to my mobile number
-      expect(notify_client).to have_received(:send_sms).with({
-        phone_number: user.mobile_number,
-        template_id: ENV["NOTIFY_OTP_VERIFICATION_TEMPLATE"],
-        personalisation: {otp: otp_at_time_of_login}
-      })
-
-      # When I enter the one-time password before it expires
-      travel 3.minutes do
-        fill_in "Please enter your six-digit verification code", with: otp_at_time_of_login
+        # And I enter an incorrect OTP
+        fill_in "Please enter your six-digit verification code", with: "000000"
         click_on "Continue"
+
+        # Then I should not be logged in
+        expect(page).to have_content("Invalid two-factor verification code")
       end
+    end
 
-      # Then I should be logged in.
-      expect(page).to have_link(t("header.link.sign_out"))
-      expect(page).to have_content("Signed in successfully.")
+    context "user initially entered an incorrect mobile number" do
+      scenario "the email/password are correct but the user still needs confirm their mobile number" do
+        incorrect_number = "123456wrong"
+        # Given a user with 2FA enabled exists
+        user = create(:administrator, :mfa_enabled, mobile_number_confirmed_at: nil, mobile_number: incorrect_number)
 
-      # And at the home page
-      expect(page).to have_content("You can search by RODA, Delivery Partner, or BEIS identifier, or by the activity's title")
+        # When I log in with that user's email and password
+        visit root_path
+        log_in_via_form(user)
+
+        # But I do not receive a message despite it having been sent
+        expect(notify_client).to have_received(:send_sms).once.with({
+          phone_number: incorrect_number,
+          template_id: ENV["NOTIFY_OTP_VERIFICATION_TEMPLATE"],
+          personalisation: {otp: user.current_otp}
+        })
```

Comment:
Would notify respond with an error here? Do we handle notify errors is they exist?